src/vanilla/server.rs

```diff
@@ -104,3 +140,6 @@ impl Server {
         Ok(())
     }
 }
+
+unsafe impl Send for Server {}
+unsafe impl Sync for Server {}
```

Comment:
Any ideas for dealing with this? The issue stems from the fact that `PKey` has a `*mut _` value. From what I can tell, the C type `EVP_KEY` is perfectly safe, provided a locking function is set (which the `openssl` crate handles nicely.)